### PR TITLE
New glossary: accessible name

### PR DESCRIPTION
**Pull request by estelle**

### CODE REVIEW

**Comment by dipikabh** [comment]
While this is great information (and I'd like to add a few edits), I am concerned about the placement of this content in Glossary. The term has been used in quite a few Aria pages and seems like it should ideally be explained in a guide or a reference page where it is first used. Is there another place where we could move this content? And then, a minimal definition can be retained in Glossary.

Perhaps you've already thought about this but after this term is part of Glossary, the occurrences of the term elsewhere will need to be linked to this glossary term (maybe in a separate PR).

**Comment by estelle** [comment]
I wanted to link to a definition of "accessible name" from the HTML section in the PWA tutorial but realized there is no glossary definition or official definition to link it to. 

The glossary seems most fitting as it's a term. The first rule of ARIA is don't use ARIA if you can use HTML instead.  The only documentation place that would make sense would be with `Aria-label`, which has some explanation, but we want to avoid pointing anyone to ARIA in this case 

I have seen stupid stuff online like `<label aria-label="first name">` (the label provides the accessible name to a form control so you don't need ARIA, but if you were to use ARIA, the arial-label would be on the form control

**Comment by hamishwillee** [comment]
@estelle Sorry, I missed my assignment to this. As it is written I feel it is quite Aria-specific - I would have suggested it as a short guide topic in the Aria section. As you say, it could also be a short glossary.

**Comment by hamishwillee** [comment]
I've never used assistive technologies so I have no practical experience of how the name is actually used. That makes it hard to assess this.

I can also see that you have thought quite carefully about the structure.
My gut feeling though is that "as a guide" this would benefit from a examples of a few obvious cases. 
For example, I was wondering what this might actually look like for a few cases, such as headings:

> Assistive technologies will provide the user with the accessibility name property, which is the accessible name along with the element's role.

I also thought that it might be good to pull these two sentences together up the top somehow (it isn't so easy, because you have put the second sentence where it is for a reason). 

>An accessible name is the name of a user interface element; it is the text associated with an HTML element that provides assistive technology users with a label for the element. 
> Accessible names convey the **specific** purpose or intent of the element. This helps users understand what the element is for and how they can interact with it.

What I'm trying to get out of this is how the label differs from what you get out of HTML, say, for a particular benefit. I _think_ that the accessible name is the specific purpose of an element (hence the addition of bold above) and is delivered along with a role to provide the unique context of each element within a doc.

**Comment by estelle** [comment]
I specifically don't want to put this in ARIA because people should NOT be using ARIA to provide an accessible name unless absolutely necessary.

In the following code, "Hello" is the accessible name.

`<a href="boop.html">Hello</a>`

"Banana" is the accessible name:

`banana`

In the following code, "First name" is the accessible name.

```html
<label for='input'>First Name</label>
<input id="input" name="yesterday" type="number">
```

Notice, no ARIA was used, nor should it have been used. I don't want to direct people to ARIA we specifically want them to not use ARIA.

File: files/en-us/glossary/accessible_name/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: Accessible name
+slug: Glossary/Accessible_name
+page-type: glossary-definition
+---
+
+An **accessible name** is the name of a user interface element; it is the text associated with an HTML element that provides assistive technology users with a label for the element.
```
Comment (estelle) [inline files/en-us/glossary/accessible_name/index.md:7]:
```suggestion
An **accessible name** is the name of a user interface element; it is the text associated with an HTML element that provides assistive technology users with a label for the element.
```

File: files/en-us/glossary/accessible_name/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (hamishwillee) [inline files/en-us/glossary/accessible_name/index.md:7]: This revision seems better. Just a suggestion:

```suggestion
An **accessible name** is the text associated with an HTML element that provides the users of assistive technology with a label for the element.
```

**Review by hamishwillee — APPROVED** [review]
@estelle Now you have reworded it, it feels a lot less aria-centric, and your point that it can reasonably be a glossary seems "obvious".

Thank you for the examples. I might have included those in the doc, but you're the expert. Certainly helped me.

I've added one last comment and approved. Feel free to ignore it if you want, and merge at will.

File: files/en-us/glossary/accessible_name/index.md (hunk 2 above)
Comment (estelle) [inline files/en-us/glossary/accessible_name/index.md:7]:
```suggestion
An **accessible name** is the name of a user interface element; it is the text associated with an HTML element that provides users of assistive technology with a label for the element.
```

File: files/en-us/glossary/accessible_name/index.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+It is best to use visible text as the accessible names. Many elements, including {{HTMLElement("a")}}, {{HTMLElement("td")}} and {{HTMLElement("button")}}, get their name from their contents. Other elements get their accessible name from the content of associated elements. For some parent elements, like {{HTMLElement("fieldset")}}, {{HTMLElement("table")}}, and {{HTMLElement("figure")}}, if those elements contain a descendant {{HTMLElement("fieldset")}}, {{HTMLElement("caption")}}, or {{HTMLElement("figcaption")}}, respectively, the association is automatic. For some other elements, like {{HTMLElement("textarea")}} and {{HTMLElement("input")}}, the accessible name also comes from an associated element, the {{HTMLElement("label")}} element, but that association has to explicitly set with a `for` attribute value on the `<label>` that matches the form control's `id`. With some elements, like {{HTMLElement("img")}}, the accessible name comes from its attributes, in this case, the `alt` attribute value.
```
Comment (estelle) [inline files/en-us/glossary/accessible_name/index.md:13]:
```suggestion
It is best to use visible text as the accessible names. Many elements, including {{HTMLElement("a")}}, {{HTMLElement("td")}} and {{HTMLElement("button")}}, get their name from their contents. For example, given `<a href="foo.html">Bar</a>`, the accessible name is "Bar."

Other elements get their accessible name from the content of associated elements. For some parent elements, like {{HTMLElement("fieldset")}}, {{HTMLElement("table")}}, and {{HTMLElement("figure")}}, if those elements contain a descendant {{HTMLElement("fieldset")}}, {{HTMLElement("caption")}}, or {{HTMLElement("figcaption")}}, respectively, the association is automatic. For some other elements, like {{HTMLElement("textarea")}} and {{HTMLElement("input")}}, the accessible name also comes from an associated element, the {{HTMLElement("label")}} element, but that association has to explicitly set with a `for` attribute value on the `<label>` that matches the form control's `id`.

With some elements, like {{HTMLElement("img")}}, the accessible name comes from its attributes, in this case, the `alt` attribute value. Given `<img src="grape.jpg" alt="banana"/>`, the image's accessible name is "banana."
```

**Pull request merged**